earth/earth_test.go

```diff
@@ -0,0 +1,193 @@
+	"github.com/golang/geo/s2"
+	"github.com/google/go-units/unit"
+)
+
+func float64Eq(x, y float64) bool {
+	if x == y {
+		return true
+	}
+	if math.Abs(x) > math.Abs(y) {
+		return math.Abs(1-y/x) < 1e-14
+	}
+	return math.Abs(1-x/y) < 1e-14
+}
+
+var degreesToMeters = []struct {
+	angle  s1.Angle
+	length unit.Length
+}{
+	{-89.93201943346866 * s1.Degree, -1e7 * unit.Meter},
+	{-30 * s1.Degree, -3335853.035324518 * unit.Meter},
+	{0 * s1.Degree, 0 * unit.Meter},
+	{30 * s1.Degree, 3335853.035324518 * unit.Meter},
+	{89.93201943346866 * s1.Degree, 1e7 * unit.Meter},
+	{90 * s1.Degree, 10007559.105973555 * unit.Meter},
+	{179.86403886693734 * s1.Degree, 2e7 * unit.Meter},
+	{180 * s1.Degree, 20015118.21194711 * unit.Meter},
+	{359.72807773387467 * s1.Degree, 4e7 * unit.Meter},
+	{360 * s1.Degree, 40030236.42389422 * unit.Meter},
+	{899.3201943346867 * s1.Degree, 1e8 * unit.Meter},
+}
+
+func TestAngleFromLength(t *testing.T) {
+	for _, test := range degreesToMeters {
+		if got, want := AngleFromLength(test.length), test.angle; !float64Eq(got.Radians(), want.Radians()) {
+			t.Errorf("AngleFromLength(%v) = %v, want %v", test.length, got, want)
+		}
+	}
+}
+
+func TestLengthFromAngle(t *testing.T) {
+	for _, test := range degreesToMeters {
+		if got, want := LengthFromAngle(test.angle), test.length; !float64Eq(got.Meters(), want.Meters()) {
+			t.Errorf("LengthFromAngle(%v) = %v, want %v", test.angle, got, want)
+		}
+	}
+}
+
+func TestLengthFromPoints(t *testing.T) {
+	tests := []struct {
+		x1, y1, z1 float64
+		x2, y2, z2 float64
+		length     unit.Length
+	}{
+		{1, 0, 0, 1, 0, 0, 0 * unit.Meter},
+		{1, 0, 0, 0, 1, 0, 10007559.105973555 * unit.Meter},
+		{1, 0, 0, 0, 1, 1, 10007559.105973555 * unit.Meter},
+		{1, 0, 0, -1, 0, 0, 20015118.21194711 * unit.Meter},
+		{1, 2, 3, 2, 3, -1, 7680820.247060414 * unit.Meter},
+	}
+	for _, test := range tests {
+		p1 := s2.PointFromCoords(test.x1, test.y1, test.z1)
+		p2 := s2.PointFromCoords(test.x2, test.y2, test.z2)
+		if got, want := LengthFromPoints(p1, p2), test.length; !float64Eq(got.Meters(), want.Meters()) {
+			t.Errorf("LengthFromPoints(%v, %v) = %v, want %v", p1, p2, got, want)
+		}
+	}
+}
+
+func TestLengthFromLatLngs(t *testing.T) {
+	tests := []struct {
+		lat1, lng1, lat2, lng2 float64
+		length                 unit.Length
+	}{
+		{90, 0, 90, 0, 0 * unit.Meter},
+		{-37, 25, -66, -155, 8562022.790666264 * unit.Meter},
+		{0, 165, 0, -80, 12787436.635410652 * unit.Meter},
+		{47, -127, -47, 53, 20015118.077688109 * unit.Meter},
+	}
+	for _, test := range tests {
+		ll1 := s2.LatLngFromDegrees(test.lat1, test.lng1)
+		ll2 := s2.LatLngFromDegrees(test.lat2, test.lng2)
+		if got, want := LengthFromLatLngs(ll1, ll2), test.length; !float64Eq(got.Meters(), want.Meters()) {
+			t.Errorf("LengthFromLatLngs(%v, %v) = %v, want %v", ll1, ll2, got, want)
+		}
+	}
+}
+
+func TestAreaFromSteradians(t *testing.T) {
+	const earthArea = 6371.01 * 6371.01 * math.Pi * 4 * unit.SquareKilometer
```

Comment:
can you reference Radius value here instead of copying the 6371.01 ?

Reply:
Done